app/src/marketplace/modules/productList/actions.js

```diff
@@ -64,22 +64,38 @@ const doGetProducts = (replace: ?boolean = false, dispatch: Function, getState:
                 }
                 dispatch(updateEntities(entities))
                 dispatch(getProductsSuccess(result, data.hasMoreProducts))
+
+                return result
             },
             (error) => {
                 dispatch(getProductsFailure(error))
             },
         )
 }
 
+type GetProducts = {
+    replace?: boolean,
+    onSuccess?: Function,
+    onError?: Function,
+}
+
 // We need to define the debounced fetch here so that we have only one reference to it
 // https://gist.github.com/krstffr/245fe83885b597aabaf06348220c2fe9
-const doGetProductsDebounced = debounce(doGetProducts, 500)
+const doGetProductsDebounced = debounce((
+    { replace, onSuccess, onError }: GetProducts,
+    dispatch: Function,
+    getState: () => StoreState,
+) => (
+    doGetProducts(replace, dispatch, getState)
+        .then(onSuccess)
+        .then(onError)
```

Comment:
Won't it hit `onError` despite being resolved successfully?

Reply:
yes 🙈 was meant to be `catch`